integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonInsertIntoCommand.scala

```diff
@@ -181,7 +183,13 @@ case class CarbonInsertIntoCommand(databaseNameOp: Option[String],
       if (isNotReArranged) {
         // Re-arrange the catalog table schema and output for partition relation
         logicalPartitionRelation =
-          getReArrangedSchemaLogicalRelation(reArrangedIndex, logicalPartitionRelation)
+          if (carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable.isMV) {
+            // For MV partition table, partition columns will be at the end. Re-arrange
```

Comment:
as discussed, please add a comment here with example, so reviewers and developers will be clear why only for MV we need to handle it separately

Reply:
done